modules/nodes-containers-start-pod-safe-sysctls.adoc

```diff
@@ -0,0 +1,104 @@
+// Module included in the following assemblies:
+//
+// * nodes/containers/nodes-containers-sysctls.adoc
+
+:_content-type: PROCEDURE
+[id="nodes-starting-pod-safe-sysctls_{context}"]
+= Starting a pod with safe sysctls
+
+You can set sysctls on pods using the pod's `securityContext`. The `securityContext` applies to all containers in the same pod.
+
+Safe sysctls are allowed by default.
+
+This example uses the pod `securityContext` to set the following safe sysctls:
+
+* `kernel.shm_rmid_forced`
+* `net.ipv4.ip_local_port_range`
+* `net.ipv4.tcp_syncookies`
+* `net.ipv4.ping_group_range`
+
+[WARNING]
+====
+To avoid destabilizing your operating system, modify sysctl parameters only after you understand their effects.
+====
+
+Use this procedure to start a pod with the configured sysctl settings.
+[NOTE]
+====
+In most cases you will be modifying an existing pod definition and adding the `securityContext` spec.
+====
+
+
+.Procedure
+
+. Create a YAML file `sysctl_pod.yaml` that defines an example pod and add the `securityContext` spec, as shown in the following example:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: sysctl-example
+  namespace: default
+spec:
+  containers:
+  - name: podexample
+    image: centos
+    command: ["bin/bash", "-c", "sleep INF"]
+  securityContext:
+    sysctls:
+    - name: kernel.shm_rmid_forced
+      value: "1"
+    - name: net.ipv4.ip_local_port_range
+      value: "32770       60666"
+    - name: net.ipv4.tcp_syncookies
+      value: "0"
+    - name: net.ipv4.ping_group_range
+      value: "0           200000000"
+----
++
+[NOTE]
+====
+An example pod is created here simply to illustrate how the changes are applied.
+====
+
+. Create the pod:
++
+[source,terminal]
+----
+$ oc apply -f sysctl_pod.yaml
```

Comment:
When I apply sysctl_pod.yaml manifest the following warning appears:
```
W0616 18:00:31.447102 1349418 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "podexample" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "podexample" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "podexample" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "podexample" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
pod/sysctl-example created
```
It happens because of the latest changes in ocp4.11 security.
In order to suppers this warning the following yaml should be applied 

```
apiVersion: v1
kind: Pod
metadata:
  name: sysctl-example
  namespace: default
spec:
  containers:
  - name: podexample
    image: centos
    command: ["bin/bash", "-c", "sleep INF"]
    securityContext:
      allowPrivilegeEscalation: false
      capabilities:
        drop: ["ALL"]
  securityContext:
    runAsNonRoot: true
    seccompProfile:
      type: RuntimeDefault
    sysctls:
    - name: kernel.shm_rmid_forced
      value: "1"
    - name: net.ipv4.ip_local_port_range
      value: "32770       60666"
    - name: net.ipv4.tcp_syncookies
      value: "0"
    - name: net.ipv4.ping_group_range
      value: "0           200000000"
```

Reply:
Thanks @kononovn I did notice that, but I thought it was something specific to the environment I tested on. Thanks for letting me know it is something new in 4.11